### PR TITLE
Implements TerrainClass.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/game/engine/terrain.cpp [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+ * @brief
+ *
+ * @copyright Chronoshift is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
```
Comment (ghost) [inline src/game/engine/terrain.cpp]: Formatting. Remove this blank space here.

File: src/game/engine/cell.h [hunk 2]
```diff
@@ -122,6 +122,8 @@ class CellClass
     BOOL Load(Straw &straw);
     BOOL Save(Pipe &pipe) const;
     BOOL Has_Bib() const { return Smudge != SMUDGE_NONE && SmudgeTypeClass::As_Reference(Smudge).Is_Bib(); }
+    void Clear_Occupant_Bit(CellOccupantEnum bit) { OccupantBit &= bit; }
```
Comment (OmniBlade) [inline src/game/engine/cell.h]: This needs to be `OccupantBit &= ~bit` to clear a bit.

File: src/game/engine/terrain.cpp [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+ *
+ * @address 0x0056A734
+ */
+BOOL TerrainClass::Catch_Fire()
+{
+    if (m_OnFire || m_Crumbling || m_Type->Get_Armor() != ARMOR_NONE) {
+        return false;
+    }
+    // what are the 0xes?
+    AnimClass *aptr = new AnimClass(ANIM_BURN_BIG, Coord_Add(Sort_Y(), (unsigned)0xFFB00000));
```
Comment (OmniBlade) [inline src/game/engine/terrain.cpp]: You don't need the casts to unsigned here, hex values are taken as unsigned by default.

File: src/game/engine/terrain.cpp [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+}
+
+/**
+ * @brief
+ *
+ * @address 0x0056AAB4
+ */
+BOOL TerrainClass::Unlimbo(coord_t coord, DirType dir)
+{
+    // is Theater_Allowed right?
```
Comment (OmniBlade) [inline src/game/engine/terrain.cpp]: Looks okay, so you can remove this comment here as it shouldn't end up in the final commit.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

**Review by OmniBlade — DISMISSED** [review]

**Review by OmniBlade — DISMISSED** [review]

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**